### PR TITLE
LibJS: Do not trim whitespace from property names when they're numbers

### LINKED ISSUE #7803: LibJS: array["  12  "] is treated as array[12]
Looks like we're incorrectly stripping whitespace around property names and turning them into numbers.

### PR DESCRIPTION
**Pull request by sin-ack**
**AK: Add a way to disable the trimming of whitespace in to_\*int**

This behavior might not always be desirable, and so this patch adds a
way to disable it.

**LibJS: Do not trim whitespace from property names when they're numbers**

Fixes #7803.

### CODE REVIEW

File: AK/StringUtils.h [hunk 1]
```diff
@@ -22,6 +22,11 @@ enum class TrimMode {
     Both
 };
 
+enum class TrimNumberString {
```
Comment (trflynn89) [inline AK/StringUtils.h:25]: A bit nit-picky, but `TrimNumberString` sounds kinda weird to me. Maybe just `TrimWhitespace`?

Comment (sin-ack) [reply AK/StringUtils.h:25]: `TrimWhitespace` doesn't explain _what_ it's trimming the whitespace from, however. And `TrimWhitespaceFromNumber` is a bit of a handful to type. :^)

Comment (linusg) [reply AK/StringUtils.h:25]: Doesn't really matter, the enum can be generic and later be used for other things - `string.to_int(TrimWhitespace::No)` seems pretty clear to me 🤔

Comment (trflynn89) [reply AK/StringUtils.h:25]: Yeah, I don't think there would be any confusion on what the whitespace is being trimmed from.

**Comment by awesomekling** [comment]
Please squash the enum change into the first commit :^)

**Comment by linusg** [comment]
Please also add a simple in-tree test!

**Comment by sin-ack** [comment]
You're giving me too many yaks to shave. :^( Will do in the evening. :^)

**Pull request merged**